### ISSUE TITLE
Errormessage installing azure-cli on ubuntu-latest indicating version mismatch

### ISSUE THREAD

**Issue by edwinvandeburgt** [Azure/azure-cli#16099]
In Azure DevOps, I try to install the latest azure-cli using:

`pip install --pre azure-cli --extra-index-url https://azurecliprod.blob.core.windows.net/edge`

This worked untill today; now I get an error:

ERROR: Requested azure-cli from https://azurecliprod.blob.core.windows.net/edge/azure-cli/azure_cli-2.15.1.post1-py3-none-any.whl has different version in metadata: '2.15.1'

Is there something I need to change?

**Comment by yungezz** [comment]
hi @fengzhou-msft could you pls help to check? thanks.

**Comment by pkoushik** [comment]
I'm seeing this issue on my end as well. 

`Requested azure-cli from https://azurecliprod.blob.core.windows.net/edge/azure-cli/azure_cli-2.15.1.post1-py3-none-any.whl has different version in metadata: '2.15.1'`

**Comment by CPArtis** [comment]
I am also receiving this error when trying to install the Azure CLI on a self-hosted agent within an Azure DevOps CI/CD pipeline. I am running the same command in the YAML script as the OP ran.

**Comment by fengzhou-msft** [comment]
The latest pip 20.3 added a version check between the version in metadata and the version in wheel name. The edge build package uses the same build pipeline with production build. We only changed the version in the wheel name after it's built. We need to modify the build pipeline for edge build to modify the version in metadata as well.

As a workaround, please ping your pip to 20.2.4.

**Comment by CPArtis** [comment]
The workaround is not working for me. I'm getting the following error.

ERROR: Command errored out with exit status 1: 'c:\sw\python36\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\pidcie1\\AppData\\Local\\Temp\\pip-install-y1mjzsvg\\vsts-cd-manager\\setup.py'"'"'; __file__='"'"'C:\\Users\\pidcie1\\AppData\\Local\\Temp\\pip-install-y1mjzsvg\\vsts-cd-manager\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\pidcie1\AppData\Local\Temp\pip-record-fpe3kh15\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\sw\python36\Include\vsts-cd-manager' Check the logs for full command output.
WARNING: You are using pip version 20.2.4; however, version 20.3 is available.
You should consider upgrading via the 'c:\sw\python36\python.exe -m pip install --upgrade pip' command.

**Comment by edwinvandeburgt** [comment]
For me, the proposed solution works:

`python -m pip install --upgrade pip==20.2.4`

**Comment by spacem** [comment]
Trying to downgrade pip is not working for us. For some reason it is reverting back to the previous version.

```
2020-12-09T23:21:00.2370446Z Generating script.
2020-12-09T23:21:00.2590345Z Script contents:
2020-12-09T23:21:00.2618591Z pip install --upgrade pip==20.2.4 --pre azure-cli==2.14.0 --extra-index-url https://azurecliprod.blob.core.windows.net/edge
2020-12-09T23:21:00.3409088Z ========================== Starting Command Output ===========================
2020-12-09T23:21:00.3729341Z ##[command]"C:\windows\system32\cmd.exe" /D /E:ON /V:OFF /S /C "CALL "D:\a\_temp\4139a48c-4fd7-48cd-ac28-40f022455e30.cmd""
2020-12-09T23:21:10.6788230Z Looking in indexes: https://pypi.org/simple, https://azurecliprod.blob.core.windows.net/edge
2020-12-09T23:21:16.0661198Z Collecting azure-cli==2.14.0
2020-12-09T23:21:16.1262320Z   Downloading azure_cli-2.14.0-py3-none-any.whl (1.9 MB)
2020-12-09T23:21:39.5389762Z Collecting pip==20.2.4
2020-12-09T23:21:39.5642514Z   Downloading pip-20.2.4-py2.py3-none-any.whl (1.5 MB)
2020-12-09T23:21:39.7153779Z Collecting antlr4-python3-runtime~=4.7.2
2020-12-09T23:21:39.7303201Z   Downloading antlr4-python3-runtime-4.7.2.tar.gz (112 kB)
2020-12-09T23:21:43.1014192Z Collecting azure-appconfiguration~=1.1.1
2020-12-09T23:21:43.1108357Z   Downloading azure_appconfiguration-1.1.1-py2.py3-none-any.whl (37 kB)
2020-12-09T23:21:43.6952196Z Collecting azure-batch~=9.0
2020-12-09T23:21:43.7071320Z   Downloading azure_batch-9.0.0-py2.py3-none-any.whl (229 kB)
2020-12-09T23:21:44.3027634Z Collecting azure-cli-core==2.14.0.*
2020-12-09T23:21:44.5001801Z   Downloading https://azurecliprod.blob.core.windows.net/edge/azure-cli-core/azure_cli_core-2.14.0.post1-py3-none-any.whl (164 kB)
2020-12-09T23:21:44.5590474Z ERROR: Requested azure-cli-core==2.14.0.* from https://azurecliprod.blob.core.windows.net/edge/azure-cli-core/azure_cli_core-2.14.0.post1-py3-none-any.whl (from azure-cli==2.14.0) has different version in metadata: '2.14.0'
2020-12-09T23:21:45.5248075Z WARNING: You are using pip version 20.3; however, version 20.3.1 is available.
```

**Comment by fengzhou-msft** [comment]
The issue is fixed with #16302. Please verify with the latest version of pip.
